### PR TITLE
dag: add workflow to run builds on K8s using dag

### PR DESCRIPTION
**Pull request by imjasonh**
This creates a cluster with x86 and arm nodes, runs a build on each, and deletes the cluster.

It (attempts to) set up GKE Workload Identity to use Service Account creds to fetch the melange signing key in GCP Secret Manager, and to push built packages to GCS.

It uses GH OIDC Federation to auth to GCP.

Signed-off-by: Jason Hall <jason@chainguard.dev>

### CODE REVIEW

**Review by jdolitsky — DISMISSED** [review]

**Comment by imjasonh** [comment]
It looks like this isn't going to work until it's merged unfortunately. The ID token can't be generated for a repo from a workflow kicked off by a fork of that repo, to prevent PRs from stealing creds.

I can try to run this manually in the staging project to help debug more stuff if that helps, but it won't help debugging CI spinning up the cluster, patching stuff, etc.

File: .github/workflows/dag-push-staging.yaml
Comment (imjasonh) [inline .github/workflows/dag-push-staging.yaml:5]: This will eventually need to become `pull_request_target` so that the ID token works, but then the workflow definition is only checked out from the repo's head, and not from the PR.

**Comment by imjasonh** [comment]
I've updated this to publish packages to a wholly different GCS bucket (`gs://wolfi-dag-test-bucket`) in the staging project.

I've also updated the workflow to trigger on pushes to the `dag` branch in the repo, which means I can start testing by pushing `git push origin dag` instead of to my fork.

@kaniini I'll hold off until I've got an affirmative response, just to be safe.

**Comment by kaniini** [comment]
Looks good to me, lets do it!

**Comment by imjasonh** [comment]
here we go!

**Pull request merged**
